contents/architecture/RESTArchitecturalStyle.md

```diff
@@ -0,0 +1,226 @@
+
+- Low entry barrier: the Web should have a simple, uniform interface for everyone to use
+- Extensibility: the Web service should be able to evolve to cater to the users&#39; evolving needs
+- Distributed Hypermedia: Hypermedia is a medium for information like graphics, audio, video, plain text and hyperlinks. The Web should use hypermedia as the user interface &quot;because of its simplicity and generality&quot;. Hypermedia has &quot;the same interface&quot; which can be used &quot;regardless of the information source&quot;, and can be used to structure information flexibly. Moreover, the hyperlinks embedded can guide the reader through the information. However, since hypermedia contains information like graphics, audio etc., the amount of information to be transferred is huge. Hence, the Web should be able to transfer a large amount of information within acceptable time limits.
+- Internet Scale:
+  - Anarchic scalability: There is no central entity to control the whole system. Hence, firstly, the Web elements should be able to cope with unexpected load. As a result, it is not feasible for the clients to keep the knowledge about all the servers, and for the servers to keep track of the state of information at each of their clients. It is also costly to keep &quot;back-pointers&quot; inside the hypermedia data elements to refer back to the elements which reference them, as potentially there can be a huge number of references. Secondly, the Web elements should be able to ensure security against malformed or maliciously structured, which requires the Web to be able to &quot;communicate authentication data and authorization controls&quot;.
+  - Independent Deployment: to cater to the needs of different users, the Web has to allow the co-existence of both old and new implementations of the Web elements and undergo &quot;gradual and fragmented change&quot;. Hence the design of the Web should be able to ease the deployment of the Web elements &quot;in a partial, iterative fashion, since it is not possible to force deployment in an orderly manner&quot;.
+
+**REST architectural style**
+
+Fielding defined architectural style as &quot;a coordinated set of architectural constraints that restricts the roles/features of architectural elements and the allowed relationships among those elements within any architecture that conforms to that style&quot;.
+
+After surveying some common architectures for network-based applications on how the architectural constraints induce their corresponding architectural properties, Fielding came up with a new architectural style – Representational State Transfer, which satisfies the requirements of the Web development.
+
+The table below describes the REST architectural constraints and the architectural properties these constraints induce. Detailed information about the architectural properties can be found in **Appendix**.
+
+<table>
```

Comment:
A few comments:
1) Why has the order of the constraints changed? Why not keep them in accordance with the order established in the previous section for easier reference?
2) What does the column on "Explanation" explain? Why the different constraints exist in the Web?
3) Inconsistencies in capitalisation for "Properties induced" column

In fact, is it possible to further simplify the content in the table, so that it is more easy to digest?